### PR TITLE
fix(admin) proper support for `PUT /{entities}/{entity}/plugins/{plugin}`

### PR DESCRIPTION
**Pull request by thibaultcha**
Prior to this patch, `PUT` requests to plugins under their associated
resources (e.g. `/services/{service}/plugins/{plugin}`) would
systematically return `404`, instead of creating the plugin if it did
not exist. This was because the `before` Lapis hook attempts to retrieve
the plugin in the database before modifying it. Those endpoints are
preserved for backwards-compatibility purposes (being too nested to be
auto-generated), so that workaround is fine.

This patch ensures that we skip this logic with `PUT` requests, plus
makes sure that we set the `self.args.post` arguments according to the
`self.params` arguments (path segments with foreign UUIDs).

Thanks @gszr for the report.

### CODE REVIEW

**Comment by thibaultcha** [comment]
@gszr Will you make sure this works on your side? Thanks!

File: kong/api/routes/plugins.lua [hunk 1]
```diff
@@ -13,12 +13,15 @@ local setmetatable = setmetatable
 
 
 local get_plugin = endpoints.get_entity_endpoint(kong.db.plugins.schema)
-local put_plugin = endpoints.put_entity_endpoint(kong.db.plugins.schema)
 local delete_plugin = endpoints.delete_entity_endpoint(kong.db.plugins.schema)
 
 
 local function before_plugin_for_entity(entity_name, plugin_field)
   return function(self, db, helpers)
+    if ngx.req.get_method() == "PUT" then
```
Comment (hishamhm) [inline kong/api/routes/plugins.lua]: `kong.request.get_method`? :)

**Comment by bungle** [comment]
> . Those endpoints are preserved for backwards-compatibility purposes (being too nested to be auto-generated), so that workaround is fine

I think we might want to make this also autogenerated for all the entities. Let me explain:
- /services/{service}/plugins/

This is the level where autogenerator stops. Now this:
- /services/{service}/plugins/{plugin}

Is not autogenerated but extended. I think we could perhaps open that url just for `PUT`. My reason is that:

- /services/{service}/plugins/
- /services/{service}/plugins/{plugin}

can be considered collection endpoints when doing `PUT`. This makes it possible to do:
`PUT /services/demo/routes/demo-route`

Which is nice. I wouldn't (though I have no strong feeling about it) support entity endpoints on that path anymore (e.g. PATCH, GET), just PUT (but that is debatable). PUT would just complete the POST that we already support.

File: kong/api/routes/plugins.lua (hunk 1 above)
Comment (bungle) [reply kong/api/routes/plugins.lua]: `self.req.method`?

Comment (thibaultcha) [reply kong/api/routes/plugins.lua]: The Admin API code doesn't use `kong.request.get_method()` anywhere yet, and uses more `ngx.req.get_method()` than it uses `self.req.method` (which is actually not documented anywhere I could find in Lapis, hence my aversion against using it).

**Pull request merged**